test/validation/api/ipsec/ipsec.c

```diff
@@ -866,6 +878,38 @@ static int ipsec_send_out_one(const ipsec_test_part *part,
 	return num_out;
 }
 
+int ipsec_test_sa_update_seq_num(odp_ipsec_sa_t sa, uint32_t seq_num)
+{
+	odp_ipsec_test_sa_operation_t sa_op;
+	odp_ipsec_test_sa_param_t sa_param;
+
+	sa_op = ODP_IPSEC_TEST_SA_UPDATE_SEQ_NUM;
+	sa_param.seq_num = seq_num;
+
+	return odp_ipsec_test_sa_update(sa, sa_op, sa_param);
+}
+
+static void ipsec_pkt_seq_num_check(odp_packet_t pkt, uint32_t seq_num)
+{
+	uint32_t l3_off = odp_packet_l3_offset(pkt);
+	uint32_t l4_off = odp_packet_l4_offset(pkt);
+	odph_ipv4hdr_t ip;
+
+	odp_packet_copy_to_mem(pkt, l3_off, sizeof(ip), &ip);
+
+	if (ip.proto == ODPH_IPPROTO_ESP) {
+		odph_esphdr_t esp;
+
+		odp_packet_copy_to_mem(pkt, l4_off, sizeof(esp), &esp);
+		CU_ASSERT_EQUAL(odp_be_to_cpu_32(esp.seq_no), seq_num);
+	} else {
```

Comment:
It would be good to check that the packet indeed is AH at this point (and not e.g. UDP).

Reply:
Ack.